version.mk

```diff
@@ -0,0 +1,5 @@
+COMMIT_NUMBER=$(shell git rev-list `git rev-list --parents HEAD | egrep "^[a-f0-9]{40}$$"`..HEAD --count)
```

Comment:
This will potentially not scale very well as the commit history grows and grows. For slow moving repositories that's fine, but for ones with faster velocity, it could be a problem.

Reply:
I've been scratching my head at this one. Is this supposed to be the number of commits since the repo's inception? Or the number of commits since the last version bump? The inner command returns no output for me with exit 1, which then results in an outer command of `git rev-list ..HEAD --count` which returns 0 always.

Reply:
I believe the former, commits since inception, not a specific `major,minor` pairing.

Reply:
Commits since inception.  Might be the `$$` which is escaping `$` for the makefile.

```
nmalik@nmalik (makefile-and-version) managed-prometheus-exporter-initcontainer$ git remote -v | grep origin
origin  git@github.com:openshift/managed-prometheus-exporter-initcontainer.git (fetch)
origin  git@github.com:openshift/managed-prometheus-exporter-initcontainer.git (push)

nmalik@nmalik (makefile-and-version) managed-prometheus-exporter-initcontainer$ git rev-list --parents HEAD | egrep "^[a-f0-9]{40}$"
a14b2d551ee1f77a13b7640828f507b029961b26

nmalik@nmalik (makefile-and-version) managed-prometheus-exporter-initcontainer$ git rev-list a14b2d551ee1f77a13b7640828f507b029961b26..HEAD --count
6
```